File: TypeScript/src/accounts/student.ts
import { AccountData } from "./accountdata";

class Student {
	
    private accountData: AccountData;
    private gpa: number;

    constructor(accountData: AccountData, gpa: number) {
        this.accountData = accountData;
        this.gpa = gpa;
    }

    public getGPA(): number {
        return this.gpa;
    }

    public updateGPA(newGPA: number) {
        this.gpa = newGPA;
    }
}

File: Java/src/main/java/account/Student.java
package account;

public class Student {

    private AccountData accountData;
    private double gpa;

    public Student(AccountData accountData, double gpa) {
        this.accountData = accountData;
        this.gpa = gpa;
    }

    public void updateGPA(double gpa) {
        this.gpa = gpa;
    }

    public double getGPA() {
        return this.gpa;
    }

}
